File: python/leetcode/contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        seen = {}
        for i in range(len(nums)):
            if nums[i] in seen:
                return True
            seen[nums[i]] = 1
        return False

File: java/hackerrank/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> seen = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (seen.getOrDefault(nums[i], 0) > 0) { return true; }
            seen.put(nums[i], 1);
        }
        return false;
    }
}